shopify_function_macro/src/lib.rs

```diff
@@ -114,6 +115,169 @@ pub fn shopify_function(
     gen.into()
 }
 
+#[derive(Clone, Default)]
+struct ShopifyFunctionTargetArgs {
+    export: Option<LitStr>,
+    query_path: Option<LitStr>,
+    schema_path: Option<LitStr>,
+    input_stream: Option<Expr>,
+    output_stream: Option<Expr>,
+}
+
+impl ShopifyFunctionTargetArgs {
+    fn parse<K: syn::parse::Parse, V: syn::parse::Parse>(
+        input: &ParseStream<'_>,
+    ) -> syn::Result<V> {
+        input.parse::<K>()?;
+        input.parse::<Token![=]>()?;
+        let value: V = input.parse()?;
+        if input.lookahead1().peek(Token![,]) {
+            input.parse::<Token![,]>()?;
+        }
+        Ok(value)
+    }
+}
+
+impl Parse for ShopifyFunctionTargetArgs {
+    fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
+        let mut args = Self::default();
+        while !input.is_empty() {
+            let lookahead = input.lookahead1();
+            if lookahead.peek(kw::export) {
+                args.export = match Self::parse::<kw::export, LitStr>(&input) {
+                    Ok(export) => Some(export),
+                    _ => None,
+                };
```

Comment:
Dumb Rust question: If there's an `export` keyword, why do we need to match like this? Could we just use `Some(...?)` like below and let an error bubble up instead of swallowing it?

Reply:
Good shout! 

```suggestion
                args.export = Some(Self::parse::<kw::export, LitStr>(&input)?);
```

Reply:
To be clear, my previous suggestion implements @andrewhassan’s suggestion of _not_ swallowing the error, which is different behavior from the original code, but arguably bubbling the error is the right thing to do?

But even if want to swallow the error there is a shorter way:
```suggestion
                args.export = Self::parse::<kw::export, LitStr>(&input).ok();
```